src/clj/puppetlabs/puppetserver/cli/subcommand.clj

```diff
@@ -15,6 +15,26 @@
         required [:config]]
     (ks/cli! cli-args specs required)))
 
+(defn environment
+  "Return a map representing the environment suitable for use with Ruby
+  subcommands.  Environment variables puppetserver needs are enforced while
+  all other variables are preserved.  This function manages GEM_HOME,
+  GEM_PATH, RUBYOPT, RUBY_OPTS, and RUBYLIB.
+
+  If no initial-env is provided then System/getenv is used.
+
+  The underlying principle is that a process should preserve as much of the
+  environment as possible in an effort to get out of the end users way.  This
+  includes PATH which should be fully managed elsewhere, e.g. the service
+  management framework."
```

Comment:
I think if we're going to go with this approach, we should consider doing it for the production code and not just the subcommands.  That way running the subcommands will get you more consistent results with what you'd get if you ran the full production stack.

PATH is an interesting one.  If we presume that the PATH variable is setup such that some system Ruby scripts suitable for running under MRI would be picked up but some JRuby equivalents would be more appropriate, I've been wondering if we'd need to account for that with a special new TK config setting like we do for GEM_HOME to make sure the environments don't mix.  I'm okay with just preserving it in the context of the subcommands for now but we may want to think through this more later. 

Would like to get @cprice404's opinion on this whole approach.